builtin/providers/cloudstack/resource_cloudstack_template.go

```diff
@@ -219,9 +236,16 @@ func resourceCloudStackTemplateRead(d *schema.ResourceData, meta interface{}) er
 	d.Set("password_enabled", t.Passwordenabled)
 	d.Set("is_ready", t.Isready)
 
-	setValueOrUUID(d, "os_type", t.Ostypename, t.Ostypeid)
-	setValueOrUUID(d, "zone", t.Zonename, t.Zoneid)
+	setValueOrUUID(d, "project", t.Project, t.Projectid)
 
+	if t.Zoneid == "" {
+		setValueOrUUID(d, "zone", t.Zonename, UnlimitedResourceID)
```

Comment:
hey @hany, in this case, the zoneName will always be empty. So for this to work, it should be:
setValueOrUUID(d, "zone", UnlimitedResourceID, UnlimitedResourceID)
That way the state will be recorded correctly, and reapplying the terraform will not destroy and recreate the template